roles/rhel_subscribe/tasks/main.yml

```diff
@@ -30,7 +30,14 @@
   redhat_subscription:
     username: "{{ rhel_subscription_user }}"
     password: "{{ rhel_subscription_pass }}"
-    autosubscribe: yes
+
+- name: Retrieve the OpenShift Pool ID
+  command: subscription-manager list --available --matches="OpenShift Enterprise, Premium*" --pool-only
```

Comment:
We should probably have the Product name to match be configurable, there will be some that will want to use `OpenShift Enterprise, Standard`, demo subscriptions, beta subscriptions and even employee subscriptions instead.

Otherwise, I see no issues with handling the pool attachment outside of the module, since it is a bug in ansible < 2.0 and even when the pool subscriptions work, if given a regex it will subscribe to all of them rather than a single sub that matches.